### ISSUE TITLE
Union method on profiles

### ISSUE THREAD

**Issue by wynged** [hypar-io/Elements#146]
I'd like to perform geometry unions on multiple profiles.  The profiles are different than polygons, because polygons don't allow holes within them. 
I've tried to make a method do this, but for some reason it only preserves the last "hole"
here's my union method.  
```c#
        public Profile Union(Profile other) {
            var clipper = new ClipperLib.Clipper();
            clipper.AddPath(this.Perimeter.ToClipperPath(), PolyType.ptSubject, true);
            clipper.AddPath(other.Perimeter.ToClipperPath(), PolyType.ptSubject, true);

            clipper.AddPaths(this.Voids.Select(p => p.ToClipperPath()).ToList(), PolyType.ptClip, true);
            clipper.AddPaths(other.Voids.Select(p => p.ToClipperPath()).ToList(), PolyType.ptClip, true);

            var solution = new List<List<ClipperLib.IntPoint>>();
            clipper.Execute(ClipperLib.ClipType.ctUnion, solution, PolyFillType.pftEvenOdd, PolyFillType.pftEvenOdd);
            var polys = solution.Select(s => s.ToPolygon()).ToArray();
            var solutionProfile = new Profile(polys[0], polys.Skip(1).ToArray());
            return solutionProfile;
}
```
And here is my test. 
```c#
        [Fact]
        public void ProfileUnion()
        {
            var circle = Polygon.Circle(3,4);
            var seed = new Profile( circle );
            for(int i=0; i<3; i++) {
                for(int j=0; j<2; j++) {
                    Transform t = new Transform(5*i, 5*j, 0);
                    var newCircle = new Profile( new Polygon(circle.Vertices.Select(v => t.OfPoint(v)).ToArray()) );
                    seed = seed.Union(newCircle);
                    if(seed == null) throw new Exception("Making union failed");
                }
            }
            Assert.Equal(2, seed.Voids.Count());
        }
```
Walking through the execution you can see that the first hole is created, but on the very next union operation, the hole is "filled".
A test method reveals that only on hole is preserved.  There should be a matching diamond hole.
![image](https://user-images.githubusercontent.com/5872187/58971169-d9bf3680-8788-11e9-8a4d-7a795956e862.png)

**Comment by hummingburd** [comment]
I suspect you might have to look at the PolyFillType parameter: 
http://www.angusj.com/delphi/clipper/documentation/Docs/Units/ClipperLib/Classes/Clipper/Methods/Execute.htmhttp://www.angusj.com/delphi/clipper/documentation/Docs/Units/ClipperLib/Types/PolyFillType.htm

**Comment by wynged** [comment]
I've resolved this issue and issues a pull request #181.  The pull request has some problems that need to be resolved I know, but I think we should continue the work there and close this issue.
The problem I was having was that once a profile gets a hole through a series of unions you need to start explicitly including this hole when doing all future unions.  So the union process is actually first to union the perimeter polygons, and then clip using all holes, form the individual profiles as well as the created holes from previous union operations.
